dask/dataframe/io/tests/test_demo.py

```diff
@@ -64,3 +65,15 @@ def test_daily_stock():
     assert isinstance(df, dd.DataFrame)
     assert 10 < df.npartitions < 31
     assert_eq(df, df)
+
+
+def test_make_timeseries_keywords():
+    df = dd.demo.make_timeseries('2000', '2001', {'A': int, 'B': int, 'C': str},
+                                 freq='1D', partition_freq='6M', A_lam=1000000, B_lam=2)
+    a_cardinality = df.A.nunique()
+    b_cardinality = df.B.nunique()
+
+    aa, bb = dask.compute(a_cardinality, b_cardinality, scheduler='single-threaded')
+
+    assert 100 < aa <= 10000000
+    assert 1 < bb <= 100
```

Comment:
Could you add a test where the column includes an underscore? I believe that your splitting logic is correct, but it took me a minute of thinking to get there.

Reply:
Done